Syncing with master

dmatej opened the pull request:
* Done also some changes which will be different between 7 and 8 (and 8.jdk17) - checks for the JVM version.

OndroMih committed:
GlassFish API - add to the Application Development guide

Add a section about GlassFish API.
Automatically generate numbers for main chapters.

OndroMih committed:
GlassFish API - add to the Application Development guide

Add a section about GlassFish API.
Automatically generate numbers for main chapters.

OndroMih committed:
Website update: Docker and IDE docs, contacts

OndroMih committed:
Website update: Docker and IDE docs, contacts

Thihup committed:
Fixes #25085 - Use HTTPS in Oracle links of the documentation

Signed-off-by: Thiago Henrique Hüpner <thihup@gmail.com>

Thihup committed:
Fixes #25085 - Use HTTPS in Oracle links of the documentation

Signed-off-by: Thiago Henrique Hüpner <thihup@gmail.com>

dmatej committed:
Merge pull request #25084 from OndroMih/ondromih-website-updates

Website update: Docker and IDE docs, contacts

dmatej committed:
Merge pull request #25087 from Thihup/issue/25085

Fixes #25085 - Use HTTPS in Oracle links of the documentation

dmatej committed:
Replaced parsing of java.version system property with usage of Runtime.version

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Replaced parsing of java.version system property with usage of Runtime.version

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Added (experimental) support for JDK 22 and JDK 23

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Added (experimental) support for JDK 22 and JDK 23

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Deleted JDK class

- was obsoleted and did not work well with JDK23
- since JDK10 it is possible to get version id directly from Runtime.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Deleted JDK class

- was obsoleted and did not work well with JDK23
- since JDK10 it is possible to get version id directly from Runtime.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Added note about higher Java versions

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Added note about higher Java versions

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed error message about insufficient JDK version

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed error message about insufficient JDK version

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removed abandoned docker image build

- We have a standalone git repository for the official docker image.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removed abandoned docker image build

- We have a standalone git repository for the official docker image.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Merge pull request #25092 from dmatej/jdk23

Experimental support of JDK23 and 22 for GlassFish 7

dmatej committed:
Merge pull request #25093 from dmatej/docker

Removed abandoned docker image build

dmatej committed:
Upgraded command-security-maven-plugin to 1.0.18

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Upgraded command-security-maven-plugin to 1.0.18

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Updated readme

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

arjantijms committed:
Merge pull request #25094 from dmatej/security-plugin

Upgraded command-security-maven-plugin to 1.0.18

arjantijms committed:
Merge pull request #25090 from OndroMih/ondromih-doc-glassfish-api

Document GlassFish API in the Application Development guide

dmatej committed:
Updated check for the version number, from 11 to 21

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

arjantijms commented:
```
18:11:25  [INFO] asciidoctor: ERROR: failed to parse formatted text: See Also: <code><a href="https://www.oracle.com/us/products/ondemand/index.html</code>">[gksce</a>]<a id="read-security-publications"></a> (reason: Expected one of [^<&], '<', '&', '</' after <code><a href="https://www.oracle.com/us/products/ondemand/index.html</code>">[gksce</a>]<a id="read-security-publications"></a> at byte 140)
18:11:25  [INFO] Converted /home/jenkins/agent/workspace/_test-using-jenkinsfile_PR-25095/docs/security-guide/target/book/book.adoc
```

dmatej committed:
Switching from deprecated soruce+target to release (compiler maven plugin)

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed link in security-guide

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
The microprofile-config module requires JDK 17

- The rest of GF uses just Java 11 as required java version

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removed ldapbp module - it was not used, it was buggy

- I just fixed its build in 2022, but it was never even tested.
- The --add-export instruction is in collision with the <release> element set
  for compiler. Obviously it would cause problems in the future.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Microprofile moved to standard modules

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
GlassFish can build on JDK17+ and is executable on JDK11+

- on JDK11 is missing the Microprofile support

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej commented:
Lets wait yet for #25096 ;-)

dmatej committed:
Merge pull request #25096 from dmatej/compiler-release

Using release instead of source+target, removing ldapbp module

dmatej committed:
Merge remote-tracking branch 'eclipse/master' into master-merge

# Conflicts:
#	README.md
#	appserver/appclient/client/acc/src/main/java/org/glassfish/appclient/client/AppClientFacade.java
#	appserver/ejb/ejb-container/src/main/java/com/sun/ejb/codegen/StaticRmiStubGenerator.java
#	docs/security-guide/src/main/asciidoc/running-in-secure-environment.adoc
#	nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/AdminMain.java
#	nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/MainHelper.java

dmatej committed:
Merge remote-tracking branch 'eclipse/master' into master-merge

# Conflicts:
#	README.md
#	appserver/ldapbp/pom.xml
#	nucleus/parent/pom.xml

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
No need to override compiler settings for microprofile config

- Also removed duplicit microprofile module, caused by merge

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>
